sound/soc/intel/boards/sof_sdw.c

```diff
@@ -865,7 +872,7 @@ static struct sof_sdw_codec_info codec_info_list[] = {
 				.dai_name = "rt715-aif2",
 				.dai_type = SOF_SDW_DAI_TYPE_MIC,
 				.dailink = {SDW_UNUSED_DAI_ID, SDW_DMIC_DAI_ID},
-				.init = sof_sdw_rt715_init,
+				.rtd_init = rt715_rtd_init,
```

Comment:
I couldn't follow this patch. In some cases, .init is removed and seems to be replaced by .rtd_init, but...

Reply:
[codec_info_list[codec_index].dais[dai_index].init](https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/intel/boards/sof_sdw.c#L1318) is called when the machine driver is creating dai links. And the `.rtd_init` will be called in the dai_link .init callback. And some codec_info_list `.init` functions only does one thing which is assign the `dai_links->init` function.
Like
``
int sof_sdw_rt715_init(struct snd_soc_card *card,
		       const struct snd_soc_acpi_link_adr *link,
		       struct snd_soc_dai_link *dai_links,
		       struct sof_sdw_codec_info *info,
		       bool playback)
{
	dai_links->init = rt715_rtd_init;

	return 0;
}
```
And the rtd_init is already called by `sof_sdw_rtd_init`. So the .init function is not needed any more.
Let me separate this to two patches. It may be easier to review.